vsintegration/src/FSharp.Editor/Formatting/BraceMatchingService.fs

```diff
@@ -14,28 +13,33 @@ type internal FSharpBraceMatchingService
         checkerProvider: FSharpCheckerProvider,
         projectInfoManager: FSharpProjectOptionsManager
     ) =
-
     
     static let defaultUserOpName = "BraceMatching"
 
-    static member GetBraceMatchingResult(checker: FSharpChecker, sourceText, fileName, parsingOptions: FSharpParsingOptions, position: int, userOpName: string) = 
+    static member GetBraceMatchingResult(checker: FSharpChecker, sourceText, fileName, parsingOptions: FSharpParsingOptions, caretPosition: int, userOpName: string) = 
         async {
             let! matchedBraces = checker.MatchBraces(fileName, sourceText.ToString(), parsingOptions, userOpName)
-            let isPositionInRange range = 
+            let isPositionInRange range =
                 match RoslynHelpers.TryFSharpRangeToTextSpan(sourceText, range) with
                 | None -> false
                 | Some range ->
-                    let length = position - range.Start
-                    length >= 0 && length <= range.Length
+                    // We need to see if the caret is contained within or on the outside of the span
+                    // that we get back from the language service.
+                    //
+                    // Ex: let x = ((12))^
+                    //
+                    // The caret can be on the outside of the last paren, but this is actually 1 position
+                    // further to the right than the end of the span that we get back.
+                    range.Contains(caretPosition) || range.End = caretPosition + 1
```

Comment:
Shouldn't this be `range.End = caretPosition`?

Reply:
That's what I originally thought, but it it results in the same behavior as #3794 (and the current implementation).

Changing to just `range.Contains(caretPosition)` fixes #3794 and also results the same behavior as the current implementation for #2092 (all cases except for `let (| A9 |)` and the caret being _inside_ a brace pair result in highlighting).

Unfortunately, debugging this is a real mess.  I think it has something to do with debugging CEs, but `isPositionInRange` is called twice as often as I would expect when debugging.